test/recipes/91-test_pkey_check.t

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/env perl
+# Copyright 2017-2020 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use strict;
+use warnings;
+
+use File::Spec;
+use OpenSSL::Test qw/:DEFAULT data_file/;
+use OpenSSL::Test::Utils;
+
+sub check_key {
+    my $f = shift;
+
+    return run(app(['openssl', 'pkey', '-check', '-text',
+                    '-in', $f]));
+}
+
+sub check_key_notok {
+    my $f = shift;
+    my $str = "$f should fail validation";
+
+    $f = data_file($f);
+
+    if ( -s $f ) {
+        ok(!check_key($f), $str);
+    } else {
+        diag("Missing file $f");
+        ok(0, $str);
+    }
+}
+
+setup("test_pkey_check");
+
+my @tests = ();
+
+push(@tests, qw(
+    ec_p256_bad_0.pem
```

Comment:
Maybe comment on what is actually 'bad' for each of these files.

Reply:
See bfaaae1a47a99dcdd8bb947fb7d991cce27446e5 and 6efa12eb6f1305d6f78ac6f5096da22aa78221e6